.github/workflows/kinds.yaml

```diff
@@ -0,0 +1,65 @@
+name: Kinds
+
+on:
+  pull_request:
+    paths:
+      - 'kinds/**'
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: kinds/go.mod
+          cache-dependency-path: kinds/*.sum
+
+      - name: Fmt
+        run: make -C kinds fmt
+
+      - name: Vet
+        run: make -C kinds vet
+
+      - name: Generate
+        run: make -C kinds generate
+
+      - name: Check generated code to be consistent
+        run: |
+          git diff --exit-code --name-only
+          if [ $? -eq 0 ]; then
+            echo "Generated files are up to date"
+          else
+            echo "Generated files are out of date"
+            exit 1
+          fi
+
+  test:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: kinds/go.mod
+          cache-dependency-path: kinds/*.sum
+
+      - name: Unit tests
+        run: make -C kinds unit-tests
+
+  validate-success:
```

Comment:
TODO: add branch protection

Reply:
this might need to be a different name, not sure (current 'ci' workflow also has a 'validate-success' job)

Reply:
Ill follow up with another pr if it does